client/client_update_test.go

```diff
@@ -148,6 +125,28 @@ func (d *testReadCloser) Close() error {
 
 func TestParseUpdateResponse(t *testing.T) {
 
+	var updateTest = []struct {
+		responseStatusCode    int
+		responseBody          []byte
+		shoulReturnError      bool
+		shouldCheckReturnCode bool
+		returnCode            int
+	}{
+		{200, []byte(correctUpdateResponse), false, true, http.StatusOK},
+		{200, []byte(correctUpdateResponseMultipleDevices), false, true, http.StatusOK},
+		// {200, []byte(updateResponseEmptyDevices), true, false, 0},
+		{204, []byte(""), false, true, http.StatusNoContent},
+		{404, []byte(`{
+		 "error": "Not found"
+		 }`), true, true, http.StatusNotFound},
+		{500, []byte(`{
+		"error": "Invalid request"
+		}`), true, false, 0},
+		// {200, []byte(malformedUpdateResponse), true, false, 0},
```

Comment:
These are obviously not valid anylonger, as the validation has been moved to `mender.go`. If you are happy with the move, then I will re-implement the tests there.

Reply:
Now that I understand it better, I think I vote against this move. The validation of the response really should be done in `client`, that's where it belongs.

I think that validating the response, and validating the Update Control map do not need to happen in the same place. Validating the response is just making sure that it follows the REST API. Whether it's actually a valid map can be checked afterwards in `mender.go`, as part of processing the response it got.